docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,437 @@
+
+This proposal addresses a subset of these problems as part of a first iteration by designing a way for Cluster API to orchestrate an external package management tool for add-on management. An external package management tool is expected to provide all the foundational capabilities for add-on management like add-on packages and repositories, templating/configuration, and management of the creation, upgrade, and deletion workflow etc.
+
+### ClusterAddonProvider Design
+
+When discussing the design of a ClusterAddonProvider, ClusterAddonProvider for Helm will be used as an example of a concrete implementation.
+
+#### ClusterAddonProvider for Helm
+
+ClusterAddonProvider for Helm proposes two new CRDs named HelmChartProxy and HelmReleaseProxy. HelmChartProxy is used to specify a Helm chart with configuration values and a selector identifying the Clusters the add-on applies to. HelmReleaseProxy contains information about the underlying Helm release installed on a cluster from a Helm chart. All the configurations will be done from the HelmChartProxy and the HelmReleaseProxy is simply used to maintain an inventory as well as surface information. Note that the specific fields in both CRDs are subject to change as the project evolves.
+
+This is a link to the prototype for [ClusterAddonProvider for Helm](https://github.com/Jont828/cluster-api-addon-provider-helm), and a demo can be found in this [Cluster API Office Hours recording](https://www.youtube.com/watch?v=dipR1Tzh4jE). The prototype implements HelmChartProxy and HelmReleaseProxy as well as controllers for both CRDs.
+
+#### HelmChartProxy
+
+HelmChartProxy serves as the user interface for defining and configuring Helm charts as well as selecting which workload clusters it will be installed on. 
+
+```go
+// HelmChartProxySpec defines the desired state of HelmChartProxy.
+type HelmChartProxySpec struct {
+	// ClusterSelector selects Clusters with a label that matches the specified key/value pair. The Helm chart will be
+	// installed on all selected Clusters. If a Cluster is no longer selected, the Helm release will be uninstalled.
+	ClusterSelector ClusterSelectorLabel `json:"clusterSelector"`
+
+	// ChartName is the name of the Helm chart in the repository.
+	ChartName string `json:"chartName,omitempty"`
+
+	// RepoURL is the URL of the Helm chart repository.
+	RepoURL string `json:"repoURL,omitempty"`
+
+	// ReleaseName is the release name of the installed Helm chart. If it is not specified, a name will be generated.
+	// +optional
+	ReleaseName string `json:"releaseName,omitempty"`
+
+	// Version is the version of the Helm chart. If it is not specified, the latest version will be used.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// Namespace is the namespace the Helm release will be installed on each selected Cluster. If it is not specified, the default namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
I recommend we rename this to `ReleaseNamespace` for two reasons: First, "namespace" alone does not make it clear that this is the release namespace; second, it would be consistent with `ReleaseName`, above.